apps/iiif/manifests/export.py

```diff
@@ -86,3 +101,367 @@ def get_zip(self, manifest, version):
         zf.close() # flush zipfile to byte stream
 
         return byte_stream.getvalue()
+
+
+class JekyllSiteExport(object):
+    def __init__(self, manifest, version, page_one=None, update_callback=None,
+                 include_images=False, deep_zoom='hosted', github_repo=None):
+        # self.volume = volume
+        self.manifest = manifest
+        self.version = version
+        # self.tei = tei
+        # self.page_one = page_one
+        self.update_callback = update_callback
+        # self.include_images = include_images
+        # self.deep_zoom = deep_zoom
+        # self.include_deep_zoom = (deep_zoom == 'include')
+        # self.no_deep_zoom = (deep_zoom == 'exclude')
+        # self.github_repo = github_repo
+
+        # # initialize github connection values to None
+        # self.github = None
+        # self.github_username = None
+        # self.github_token = None
+
+
+    def log_status(self, msg):
+        logger.debug(msg)
+        if self.update_callback is not None:
+            self.update_callback(msg)
+
+    def notify_msg(self, msg):
+        log_status(msg)
+
+
+    def get_zip(self):
+        self.log_status("DEADBEEF")
+        return self.volume_export()
+
+    def import_iiif_jekyll(self, manifest, tmpdir):
+        # run the script to get a freash import of tei as jekyll site content
+        self.log_status('Running jekyll import IIIF manifest script')
+        # jekyllimport_tei_script = settings.JEKYLLIMPORT_TEI_SCRIPT
+        jekyllimport_manifest_script = settings.JEKYLLIMPORT_MANIFEST_SCRIPT
+        import_command = [jekyllimport_manifest_script, '-q', manifest.baseurl+'/manifest', tmpdir]
+        # TODO
+        # # if a page number is specified, pass it as a parameter to the script
+        # if self.page_one is not None:
+        #     import_command.extend(['--page-one', unicode(self.page_one)])
+        # # if no deep zoom is requested, pass through so the jekyll
+        # #  config can be updated appropriately
+        # if self.no_deep_zoom:
+        #     import_command.append('--no-deep-zoom')
+
+        try:
+            logger.debug('Jekyll import command: %s', ' '.join(import_command))
+            subprocess.check_call(' '.join(import_command), shell=True)
+#            subprocess.check_call(import_command, cwd=tmpdir)
+        except subprocess.CalledProcessError:
+            err_msg = 'Error running jekyll import on IIIF manifest'
+            if self.update_callback is not None:
+                self.update_callback(err_msg, 'error')
+            raise ExportException(err_msg)
+
+
+
+    def generate_website(self):
```

Comment:
Is there a need to background this process. I know we are going to skip the socket to update user on progress. Though I do think it is helpful. I think the lack of interest in providing feedback is rooted in the fact it was really flakey in v1. Maybe polling would work better?